bin/homeshick

```diff
@@ -182,7 +183,13 @@ case $cmd in
         refresh)
           (refresh $threshhold "$param") ;;
         pull)
-          (pull "$param") && pull_completed+=("$param") ;;
+          prev_hash=$(cd "$repos/$param" && git rev-parse HEAD)
```

Comment:
At work we pre-initialize the castles with `git init <castle>; git remote add origin ...` when provisioning servers and put a `homeshick pull --batch; homeshick link -f; exec $SHELL` in an otherwise empty `.bashrc`. This way the dotfiles are initialized on first login. However, in that case `HEAD` does not exist and `rev-parse` will fail.  
Maybe just bail when `rev-parse` fails and assume the `link` call is handled in another way.

Reply:
True, though maybe we should silence the error?

Reply:
I can't get that pre-initialization to work. After running those commands in a clean shell, I get this error message on the current master:
> **no upstream** Could not pull dotfiles, it has no upstream

Reply:
Here's the actual commands we're running. I believe the last line is what does the trick.
```
git init $HOME/.homesick/repos/dotfiles
(
  cd $HOME/.homesick/repos/dotfiles
  git remote add origin git@github.com:user/dotfiles
  printf '[branch "master"]\n  remote = origin\n  merge = refs/heads/master' >> .git/config)
)
```